core/services/ocr2/plugins/ccip/internal/cache/commit_roots_test.go

```diff
@@ -1,247 +1,296 @@
+		createReportAcceptedLog(t, chainID, commitStoreAddr, 5, 1, root5, block4),
+	}
+	require.NoError(t, orm.InsertLogsWithBlock(ctx, inputLogs, logpoller.NewLogPollerBlock(utils.RandomBytes32(), 5, time.Now(), 3)))
+	roots, err = rootsCache.RootsEligibleForExecution(ctx)
+	require.NoError(t, err)
+	assertRoots(t, roots, root2, root3, root4, root5)
+
+	// Mark root in the middle as executed but keep the oldest one still waiting
+	rootsCache.MarkAsExecuted(root3)
+	roots, err = rootsCache.RootsEligibleForExecution(ctx)
+	require.NoError(t, err)
+	assertRoots(t, roots, root2, root4, root5)
+
+	// Simulate reorg by removing all unfinalized blocks
+	require.NoError(t, orm.DeleteLogsAndBlocksAfter(ctx, 4))
+	roots, err = rootsCache.RootsEligibleForExecution(ctx)
+	require.NoError(t, err)
+	assertRoots(t, roots, root2)
+
+	// Root4 comes back but in the
+	inputLogs = []logpoller.Log{
+		createReportAcceptedLog(t, chainID, commitStoreAddr, 4, 1, root4, block25),
+	}
+	require.NoError(t, orm.InsertLogsWithBlock(ctx, inputLogs, logpoller.NewLogPollerBlock(utils.RandomBytes32(), 5, time.Now(), 3)))
+	roots, err = rootsCache.RootsEligibleForExecution(ctx)
+	require.NoError(t, err)
+	assertRoots(t, roots, root2, root4)
+
+	// Mark everything as executed
+	rootsCache.MarkAsExecuted(root2)
+	rootsCache.MarkAsExecuted(root4)
+	roots, err = rootsCache.RootsEligibleForExecution(ctx)
+	require.NoError(t, err)
+	assertRoots(t, roots)
 }
 
-func TestEvictingElements(t *testing.T) {
-	c := newCommitRootsCache(logger.TestLogger(t), 1*time.Hour, 1*time.Millisecond, 1*time.Millisecond, 1*time.Millisecond)
+func Test_RootsEligibleForExecutionWithReorgs(t *testing.T) {
+	ctx := testutils.Context(t)
+	chainID := testutils.NewRandomEVMChainID()
+	orm := logpoller.NewORM(chainID, pgtest.NewSqlxDB(t), logger.TestLogger(t))
+	lpOpts := logpoller.Opts{
+		PollPeriod:               time.Hour,
+		FinalityDepth:            2,
+		BackfillBatchSize:        20,
+		RpcBatchSize:             10,
+		KeepFinalizedBlocksDepth: 1000,
+	}
+	lp := logpoller.NewLogPoller(orm, nil, logger.TestLogger(t), nil, lpOpts)
 
-	k1 := [32]byte{1}
-	c.Snooze(k1)
+	commitStoreAddr := utils.RandomAddress()
 
-	time.Sleep(10 * time.Millisecond)
+	block1 := time.Now().Add(-8 * time.Hour)
+	block2 := time.Now().Add(-5 * time.Hour)
+	block3 := time.Now().Add(-1 * time.Hour)
 
-	assert.False(t, c.IsSkipped(k1))
-}
+	root1 := utils.RandomBytes32()
+	root2 := utils.RandomBytes32()
+	root3 := utils.RandomBytes32()
 
-func Test_UnexecutedRoots(t *testing.T) {
-	type rootWithTs struct {
-		root [32]byte
-		ts   time.Time
+	// Genesis block
+	require.NoError(t, orm.InsertBlock(ctx, utils.RandomBytes32(), 1, block1, 1))
+	inputLogs := []logpoller.Log{
+		createReportAcceptedLog(t, chainID, commitStoreAddr, 2, 1, root1, block1),
+		createReportAcceptedLog(t, chainID, commitStoreAddr, 2, 2, root2, block1),
+		createReportAcceptedLog(t, chainID, commitStoreAddr, 3, 1, root3, block2),
```

Comment:
These values look weird to me, if they're correct could you explain what this is doing? Looks like block 2 has 2 commit reports but the timestamp and log indices seem off, this is what I expected:
```suggestion
		createReportAcceptedLog(t, chainID, commitStoreAddr, 2, 1, root1, block2),
		createReportAcceptedLog(t, chainID, commitStoreAddr, 2, 2, root2, block2),
		createReportAcceptedLog(t, chainID, commitStoreAddr, 3, 3, root3, block3),
```

Reply:
```go
		createReportAcceptedLog(t, chainID, commitStoreAddr, 2, 1, root1, block1),
		createReportAcceptedLog(t, chainID, commitStoreAddr, 2, 2, root2, block1),
```

Block number 2 contains two logs
* log1 - root1
* log2 - root2

This is a valid situation in which two CommitReports are included in the same block